src/section/shared_state.rs

```diff
@@ -241,34 +243,39 @@ impl SharedState {
         &mut self,
         name: &XorName,
         proof: Proof,
-        recommended_section_size: usize,
```

Comment:
better still use this recommended_section_size as input.
avoid importing `Core` for this file.